src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -1940,6 +1945,34 @@ def process_declarations(self, parent, nodes, arg_list,
         for decl in walk(nodes, Fortran2003.Derived_Type_Def):
             self._process_derived_type_decln(parent, decl, visibility_map)
 
+        # INCLUDE statements are *not* part of the Fortran language and
+        # can appear anywhere. Therefore we have to do a walk to make sure we
+        # find them if they are present.
+        incl_nodes = walk(nodes, (Fortran2003.Include_Stmt,
+                                  C99Preprocessor.Cpp_Include_Stmt))
+        if incl_nodes:
+            config = Config.get()
+            # Just generate an error message for the first include we've found
+            filename = incl_nodes[0].children[0].string
+            rtxt = (f"while processing the declarations in routine "
+                    f"'{parent.name}'")
+            if isinstance(incl_nodes[0], Fortran2003.Include_Stmt):
+                msg = (
+                    f"Fortran INCLUDE statements are not supported but found "
+                    f"an include for file '{filename}' {rtxt}. This file must "
+                    f"be made available to the Fortran parser by specifying "
+                    f"its "
+                    f"location with a -I flag. (The list of directories to "
+                    f"search is currently set to: {config.include_paths}.)")
```

Comment:
This error message needs improving because Fortran include statements ARE supported (as we go on to say in the error message), is just that the link to this particular 'filename' was not found.

(This this is an user message I would not explain that fparser INCLUDE are not supported, is fparser than needs to expand them with the provided paths - this is internal details)

Reply:
Oh yes, good spot. I've attempted to improve it.